okta/resource_okta_app_oauth_api_scope.go

```diff
@@ -0,0 +1,242 @@
+package okta
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/okta/okta-sdk-golang/v2/okta"
+)
+
+func resourceAppOAuthApiScope() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAppOAuthApiScopeCreate,
+		ReadContext:   resourceAppOAuthApiScopeRead,
+		UpdateContext: resourceAppOAuthApiScopeUpdate,
+		DeleteContext: resourceAppOAuthApiScopeDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+				scopes, _, err := getOktaClientFromMetadata(m).Application.ListScopeConsentGrants(ctx, d.Id(), nil)
+				if err != nil {
+					return nil, err
+				}
+
+				_ = d.Set("app_id", d.Id())
+				if len(scopes) > 0 {
+					// Assume issuer is the same for all granted scopes, taking the first
+					_ = d.Set("issuer", scopes[0].Issuer)
+				} else {
+					return nil, errors.New("no application scope found")
+				}
+				if err = setOAuthApiScopes(d, scopes); err != nil {
+					return nil, err
+				}
+
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"app_id": {
+				Required:    true,
+				Type:        schema.TypeString,
+				Description: "ID of the application.",
+				ForceNew:    true,
+			},
+			"issuer": {
+				Required:    true,
+				Type:        schema.TypeString,
+				Description: "The issuer of your Org Authorization Server, your Org URL.",
+			},
+			"scopes": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Scopes of the application for which consent is granted.",
```

Comment:
Could you please add validation for the scopes based on the list from [here](https://developer.okta.com/docs/guides/implement-oauth-for-okta/scopes/)?

Reply:
Totally agree, validation would be good here but it looks like tf doesn't like validation on lists. I get the following error:

```
1 error occurred:
	* resource okta_app_oauth_api_scope: scopes: ValidateFunc and
ValidateDiagFunc are not yet supported on lists or sets.
```

I could put the validation in the setter function but that becomes a bit messy, do you have any idea?

Reply:
Try putting validation for each element `Elem:        &schema.Schema{Type: schema.TypeString},`

Reply:
Ha, stupid me, yes that works perfectly. Thanks again!